src/query/storage/m3/storage.go

```diff
@@ -287,7 +291,7 @@ func (s *m3storage) Write(
 		datapoint := datapoint
 		wg.Add(1)
 		s.writeWorkerPool.Go(func() {
-			if err := s.writeSingle(ctx, query, datapoint, identID, tagIter); err != nil {
+			if err := s.writeSingle(ctx, query, datapoint, id, tagIter); err != nil {
```

Comment:
Could we add a special case here for when `len(query.Datapoints) == 1` to just `return s.writeSingle(...)` as discussed?

Reply:
done